src/Twig/Extensions.php

```diff
@@ -249,6 +249,10 @@ private function initLocale()
         $this->locale = $locale;
         $this->numberFormatter = new NumberFormatter($locale, NumberFormatter::DECIMAL);
         $this->moneyFormatter = new NumberFormatter($locale, NumberFormatter::CURRENCY);
+
+        // returns always two decimal digits
+        $this->numberFormatter->setAttribute(NumberFormatter::FRACTION_DIGITS, 2);
```

Comment:
These lines are the most critical ones and I doubt if that is correct for every locale. 
You are skipping the configured ICU formatting rules and replacing them by a custom implementation. 
Even if the current system has flaws (which I am sue there are) the change to a hard coded value seems to be a breaking change that is likely wrong when you have customers is different countries with different currencies.